aea/mail/base.py

```diff
@@ -374,6 +375,8 @@ def __init__(
         self._in_queue = queue.Queue()  # type: queue.Queue
         self._out_queue = None  # type: Optional[asyncio.Queue]
 
+        self._connect_all_task = None  # type: Optional[Future]
+        self._disconnect_all_task = None  # type: Optional[Future]
```

Comment:
these two futures handle the case when some connection/disconnection are pending.